README.md
# Object-Calculator
  This program consists of 2 objects which is Calculator 1 and Calculator 2, it is followed by a Calculator Test Driver which uses the two different Calculator at once. The progress while using the calculators will be transferred immediately at 'History.txt'.

###

# Calculator Test Driver
  This program calls the objects that'll be used for performing basic operations such as addition, multiplication and division. 
  ## Flow
  1. The Calculator status will be presented.
  2. The user will be asked to asked if it wanted to use Calculator 1.
      + If yes, then the user will proceed using Calculator 1. 'Histor.txt' will be updated that the user used Calculator 1. 
     ```
     'History.txt' will print 
     Calculator 1 History.
     ```
         If not, then it will proceed using Calculator 2.
     
     + Next, the user will be asked which operation will be used.Addition, Subtraction, Multiplication and Division are the only choices. 
     ```
     Which operation ('addition', 'subtraction', 'multiplication' or 'division') would you like to use?: 
     ```
     + It will create a while loop that will repeat the operation. It can only be stopped responding 'no' when asked.
     ```
     while response == 'yes':
         Cal1.addition()
         response = input("Would you like to repeat ADDITION?('yes' or 'no'): ") 
     ```
     + When the while loop is broken, the it will repeat the process for another operation.
     + If using calculator 1 is done, it will turn off and proceed Calculator 2.
     ```
      "Turning off Calculator 1..."
     ```
     
   3. The user will be asked to asked if it wanted to use Calculator 2.
      + If yes, then the user will proceed using Calculator 2. 'Histor.txt' will be updated that the user used Calculator 2. 
      ```
      'History.txt' will print 
      Calculator 2 History.
      ```
     
         If not, Calculator 2 will turn off.
       + Next, the user will be asked which operation will be used.Addition, Subtraction, Multiplication and Division are the only choices.
        
        ```
        Which operation ('addition', 'subtraction', 'multiplication' or 'division') would you like to use?: 
        ```
       + It will create a while loop that will repeat the operation. It can only be stopped responding 'no' when asked.
       ```
        while response == 'yes':
            Cal2.addition()
            response = input("Would you like to repeat ADDITION?('yes' or 'no'): ") 
        ```
     
        + When the while loop is broken, the it will repeat the process for another operation.
        + If using calculator 2 is done, it will turn off. 
      ```
        "Turning off Calculator 2..."
      ```
   

###

# Calculator 1
  This calculator allow you to perform basic operations such as addition, subtraction, multiplication and division.It allows the user to perform basic operations with a three (3) decimal places. The progress of any operation used will be written in 'History.txt'.

## Guide

Create an 2 input that'll be involved in operations. Then use it for the operation. Make sure that the float() is applied in inputs for the process. Lastly, associate the result with "%.3f" just like on the given below:
```
def addition(calc):
  calc.input1 = input("Enter first number: ")
  calc.input2 = input("Enter second number: ")
  result = float(calc.input1) + float(calc.input2)
  
  calc.result1 = "%.3f"%result
```
This process is applied to all operation. However, The operand is changed per operation.

###
# Calculator 2
  This calculator allow you to perform basic operations such as addition, subtraction, multiplication and division. It allows the user to perform basic operations without any decimal places. However, there is a special case in this calculator. When division is used, the remainder will be stated.The progress of any operation used will be written in 'History.txt'.
 
 
## Guide
Just like the first calculator create an 2 input that'll be involved in operations. Then use it for the operation.  Use int() function on the inputs like on the given below:
```
def addition(calc):
  calc.input1 = input("Enter first number: ")
  calc.input2 = input("Enter second number: ")
  result = int(calc.input1) + int(calc.input2)
    
```
This process is applied to all operation. However, The operand is changed per operation.

There is a special case  division where in remainder is involved. Modulus (%) whould be applied from first input followed by second input.

```
def division(calc):
  calc.input1 = input("Enter first number: ")
  calc.input2 = input("Enter second number: ")
  result = int(calc.input1) / int(calc.input2)
  remainder = int(calc.input1) % int(calc.input2)
    
```

### 



History.txt

Calculator 2 History
99 / 4 = 25 Remainder 3
9 / 2 = 4 Remainder 1
100 + 3 = 103


Calculator.py
#Colors and text forms
BLUE = '\033[94m'
DARKCYAN = '\033[36m'
PURPLE = '\033[95m'
GREEN = '\033[92m'
RED = '\033[91m'
BOLD = '\033[1m'
END = '\033[0m'
BORDER = 175*'='
print(BOLD+PURPLE+BORDER+END)

#Class Calculator Default

class Calculator:
    def __init__(calc, input1 = 0,input2 = 0, result1 = 0):
        calc.input1 = input1
        calc.input2 = input2
        calc.result1 = result1

#Addition function: While loop function, File history
    def addition(calc):
        try:
            print(BOLD+PURPLE+BORDER+END)
            print('\n')
            print(BOLD+BLUE+"Enter your first number:"+END)
            calc.input1 = input(" ")
           
            print(BOLD+BLUE+"Enter your second number:"+END)
            calc.input2 = input(" ")
            result = float(calc.input1) + float(calc.input2)
    
        except ValueError:
            print('\n')
            print(BOLD + RED + "Sorry! Only integers and floats are allowed"+ END )

        except TypeError:
            print('\n')
            print(BOLD + RED + "Non-integer and non-float are cannot be used."+ END )

        calc.result1 = "%.3f"%result
        print('\n')
        print(str(calc.input1) ,"+", str(calc.input2),"=", calc.result1)
        history = open('History.txt', 'a')
        history.write(calc.input1),history.write(' + '), history.write(calc.input2), history.write(' = '), history.write(calc.result1), history.write('\n')
        
        print('\n')
        print(BOLD+PURPLE+BORDER+END)

#Subtraction function: WHile loop function
    def subtraction(calc):
        try:
            print(BOLD+PURPLE+BORDER+END)
            print('\n')

            print(BOLD+BLUE+"Enter your first number:"+END)
            calc.input1 = input(" ")
           
            print(BOLD+BLUE+"Enter your second number:"+END)
            calc.input2 = input(" ")
            result = float(calc.input1) - float(calc.input2)
    
        except ValueError:
            print('\n')
            print(BOLD + RED + "Sorry! Only integers and floats are allowed"+ END )

        except TypeError:
            print('\n')
            print(BOLD + RED + "Non-integer and non-float are cannot be used."+ END )
    
        calc.result1 = "%.3f"%result
        print('\n')
        print(str(calc.input1) ,"-", str(calc.input2),"=", calc.result1)
        history = open('History.txt', 'a')
        history.write(calc.input1),history.write(' - '), history.write(calc.input2), history.write(' = '), history.write(calc.result1), history.write('\n')

        print('\n')
        print(BOLD+PURPLE+BORDER+END)
            
#Multiplication function: While loop function
    def multplication(calc):
        try:
            print(BOLD+PURPLE+BORDER+END)
            print('\n')

            print(BOLD+BLUE+"Enter your first number:"+END)
            calc.input1 = input(" ")
           
            print(BOLD+BLUE+"Enter your second number:"+END)
            calc.input2 = input(" ")
            result = float(calc.input1) * float(calc.input2)
    
        except ValueError:
            print('\n')
            print(BOLD + RED + "Sorry! Only integers and floats are allowed"+ END )

        except TypeError:
            print('\n')
            print(BOLD + RED + "Non-integer and non-float are cannot be used."+ END )

        calc.result1 = "%.3f"%result
        print('\n')
        print(str(calc.input1) ,"x", str(calc.input2),"=", calc.result1)
        history = open('History.txt', 'a')
        history.write(calc.input1),history.write(' x '), history.write(calc.input2), history.write(' = '), history.write(calc.result1), history.write('\n')

        print('\n')
        print(BOLD+PURPLE+BORDER+END)

#Division function: While loop function
    def division(calc):
        try:
            print(BOLD+PURPLE+BORDER+END)
            print('\n')

            print(BOLD+BLUE+"Enter your first number:"+END)
            calc.input1 = input(" ")
           
            print(BOLD+BLUE+"Enter your second number:"+END)
            calc.input2 = input(" ")
            result = float(calc.input1) / float(calc.input2)
    
        except ValueError:
            print('\n')
            print(BOLD + RED + "Sorry! Only integers and floats are allowed"+ END )

        except TypeError:
            print('\n')
            print(BOLD + RED + "Non-integer and non-float are cannot be used."+ END )

        calc.result1 = "%.3f"%result
        print('\n')
        print(str(calc.input1) ,"/", str(calc.input2),"=", calc.result1)
        history = open('History.txt', 'a')
        history.write(calc.input1),history.write(' / '), history.write(calc.input2), history.write(' = '), history.write(calc.result1), history.write('\n')

        print('\n')
        print(BOLD+PURPLE+BORDER+END)

class Calculator_Override(Calculator):
    def __init__(calc, input1 = 0,input2 = 0, result1 = 0):
        calc.input1 = input1
        calc.input2 = input2
        calc.result1 = result1

    def addition(calc, input1, input2):
        result = float(input1) + float(input2)
        result = "%.3f"%result
        return result

    def subtraction(calc, input1, input2):
        result = float(input1) - float(input2)
        result = "%.3f"%result
        return result

    def multiplication(calc, input1, input2):
        result = float(input1) * float(input2)
        result = "%.3f"%result
        return result

    def division(calc, input1, input2):
        result = float(input1) / float(input2)
        result = "%.3f"%result
        return result  

Calculator2.py
#Colors and text forms
BLUE = '\033[94m'
DARKCYAN = '\033[36m'
PURPLE = '\033[95m'
GREEN = '\033[92m'
RED = '\033[91m'
BOLD = '\033[1m'
END = '\033[0m'
BORDER = 175*'='

print(BOLD+BORDER+END)

#Class Calculator Default

class Calculator2:
    def __init__(calc, input1 = 0,input2 = 0, result2 = 0):
        calc.input1 = input1
        calc.input2 = input2
        calc.result2 = result2

#Addition function: While loop function, File history
    def addition(calc):
        try:
            print(BOLD+BORDER+END)
            print('\n')
            print(BOLD+BLUE+"Enter your first number:"+END)
            calc.input1 = input(" ")
           
            print(BOLD+BLUE+"Enter your second number:"+END)
            calc.input2 = input(" ")
            calc.result2 = int(calc.input1) + int(calc.input2)
    
        except ValueError:
            print('\n')
            print(BOLD + RED + "Sorry! Only integers and ints are allowed"+ END )

        except TypeError:
            print('\n')
            print(BOLD + RED + "Non-integer and non-int are cannot be used."+ END )

        calc.result2 = str(int(calc.result2))
        print('\n')
        print(str(calc.input1) ,"+", str(calc.input2),"=", calc.result2)
        history = open('History.txt', 'a')
        history.write(calc.input1),history.write(' + '), history.write(calc.input2), history.write(' = '), history.write(calc.result2), history.write('\n')
        
        print('\n')
        print(BOLD+BORDER+END)

#Subtraction function: WHile loop function
    def subtraction(calc):
        try:
            print(BOLD+BORDER+END)
            print('\n')

            print(BOLD+BLUE+"Enter your first number:"+END)
            calc.input1 = input(" ")
           
            print(BOLD+BLUE+"Enter your second number:"+END)
            calc.input2 = input(" ")
            calc.result2 = int(calc.input1) - int(calc.input2)
    
        except ValueError:
            print('\n')
            print(BOLD + RED + "Sorry! Only integers and ints are allowed"+ END )

        except TypeError:
            print('\n')
            print(BOLD + RED + "Non-integer and non-int are cannot be used."+ END )
    
        calc.result2 = str(int(calc.result2))
        print('\n')
        print(str(calc.input1) ,"-", str(calc.input2),"=", calc.result2)
        history = open('History.txt', 'a')
        history.write(calc.input1),history.write(' - '), history.write(calc.input2), history.write(' = '), history.write(calc.result2), history.write('\n')

        print('\n')
        print(BOLD+BORDER+END)
            
#Multiplication function: While loop function
    def multplication(calc):
        try:
            print(BOLD+BORDER+END)
            print('\n')

            print(BOLD+BLUE+"Enter your first number:"+END)
            calc.input1 = input(" ")
           
            print(BOLD+BLUE+"Enter your second number:"+END)
            calc.input2 = input(" ")
            calc.result2 = int(calc.input1) * int(calc.input2)
    
        except ValueError:
            print('\n')
            print(BOLD + RED + "Sorry! Only integers and ints are allowed"+ END )

        except TypeError:
            print('\n')
            print(BOLD + RED + "Non-integer and non-int are cannot be used."+ END )

        calc.result2 = str(int(calc.result2))
        print('\n')
        print(str(calc.input1) ,"x", str(calc.input2),"=", calc.result2)
        history = open('History.txt', 'a')
        history.write(calc.input1),history.write(' x '), history.write(calc.input2), history.write(' = '), history.write(calc.result2), history.write('\n')

        print('\n')
        print(BOLD+BORDER+END)

#Division function: While loop function
    def division(calc):
        try:
            print(BOLD+BORDER+END)
            print('\n')

            print(BOLD+BLUE+"Enter your first number:"+END)
            calc.input1 = input(" ")
           
            print(BOLD+BLUE+"Enter your second number:"+END)
            calc.input2 = input(" ")
            calc.result2 = int(calc.input1) / int(calc.input2)
            remainder = int(calc.input1) % int(calc.input2)
            
    
        except ValueError:
            print('\n')
            print(BOLD + RED + "Sorry! Only integers and ints are allowed"+ END )

        except TypeError:
            print('\n')
            print(BOLD + RED + "Non-integer and non-int are cannot be used."+ END )

        calc.result2 = str(int(calc.result2))
        print('\n')
        print(str(calc.input1) ,"/", str(calc.input2),"=", calc.result2, "Remainder", remainder)
        history = open('History.txt', 'a')
        history.write(calc.input1),history.write(' / '), history.write(calc.input2), history.write(' = '), history.write(calc.result2), history.write(" Remainder "),history.write(remainder),  history.write('\n')

        print('\n')
        print(BOLD+BORDER+END)


class Calculator_Override2(Calculator2):
    def __init__(calc, input1 = 0,input2 = 0, result1 = 0):
        calc.input1 = input1
        calc.input2 = input2
        calc.result1 = result1

    def addition(calc, input1, input2):
        result = int(input1) + int(input2)
        result = "%.0f"%result
        return result

    def subtraction(calc, input1, input2):
        result = int(input1) - int(input2)
        result = "%.0f"%result
        return result

    def multiplication(calc, input1, input2):
        result = int(input1) * int(input2)
        result = "%.0f"%result
        return result

    def division(calc, input1, input2):
        result = int(input1) / int(input2)
        result = "%.0f"%result
        return result
    
    def div_rem(calc, input1, input2):
        result = int(input1) % int(input2)
        remainder = "%.0f"%result
        return remainder
        

Calculator Test Driver.py
#COLORS 
BLUE = '\033[94m'
DARKCYAN = '\033[36m'
PURPLE = '\033[95m'
GREEN = '\033[92m'
RED = '\033[91m'
BOLD = '\033[1m'
END = '\033[0m'
BORDER = 175*'='

from Calculator import Calculator_Override
from Calculator2 import Calculator_Override2




print(BOLD+DARKCYAN+BORDER+END)
print(BOLD+DARKCYAN+BORDER+END)
print('\n')

print("CALCULATOR STATUS:")
print('\n')
print(BOLD+GREEN+"Calculator 1 is working fine..."+END)

print(BOLD+GREEN+"Calculator 2 is working fine..."+END) 

print('\n')
print(BOLD+DARKCYAN+BORDER+END)
print(BOLD+DARKCYAN+BORDER+END)

print('\n')

print(BOLD+GREEN+"Calculator 1:"+END)
print('\n')
print("Description:")
print('\n')
print(" This calculator allow you to perform basic operations such as addition, subtraction, multiplication and division.")
print("It allows the user to perform basic operations with a three (3) decimal places.")
print("The progress of any operation used will be written in 'History.txt'.")
print('\n')
print(BOLD+DARKCYAN+BORDER+END)
print(BOLD+DARKCYAN+BORDER+END)


print('\n')

print(BOLD+GREEN+"Calculator 2:"+END)
print('\n')
print("Description:")
print('\n')
print(" This calculator allow you to perform basic operations such as addition, subtraction, multiplication and division.")
print("It allows the user to perform basic operations without any decimal places.")
print("However, there is a special case in this calculator. When division is used, the remainder will be stated.")
print("The progress of any operation used will be written in 'History.txt'.")
print('\n')
print(BOLD+DARKCYAN+BORDER+END)
print(BOLD+DARKCYAN+BORDER+END)

#while loop
Cal1 = Calculator_Override() 
response = input("Would you like to open Calculator 1?('yes' or 'no'): ")
if response == 'yes':
    with open('History.txt', 'a') as History1:
        History1.write('\n')
        History1.write("Calculator 1 History"), History1.write('\n')
    
    while response == 'yes':
        operation = input("Which operation ('addition', 'subtraction', 'multiplication' or 'division') would you like to use?: ")
        
        print('\n')
        print(BOLD+DARKCYAN+BORDER+END)
        print('\n')

        if operation.lower() == 'addition':
            input1 = input("Enter your first number:   ")
            input2 = input("Enter your second number:   ")
            print('\n')
            add = Cal1.addition(input1, input2)
            print(input1, " + ", input2, " = ", add)
            print('\n')

            history = open('History.txt', 'a')
            history.write(input1),history.write(' + '), history.write(input2), history.write(' = '), history.write(add), history.write('\n')
            history.close()
            
            response = input("Would you like to repeat ADDITION?('yes' or 'no'): ")
            

            while response == 'yes':
                print('\n')
                print(BOLD+DARKCYAN+BORDER+END)
                input1 = input("Enter your first number:   ")
                input2 = input("Enter your second number:   ")
                print('\n')
                add = Cal1.addition(input1, input2)
                print(input1, " + ", input2, " = ", add)
                
                print('\n')

                history = open('History.txt', 'a')
                history.write(input1),history.write(' + '), history.write(input2), history.write(' = '), history.write(add), history.write('\n')
                history.close()

                response = input("Would you like to repeat ADDITION?('yes' or 'no'): ") 

        elif operation.lower() == 'subtraction':
            input1 = input("Enter your first number:   ")
            input2 = input("Enter your second number:   ")

            print('\n')

            subtract = Cal1.subtraction(input1, input2)
            print(input1, " - ", input2, " = ", subtract)
            print('\n')

            history = open('History.txt', 'a')
            history.write(input1),history.write(' - '), history.write(input2), history.write(' = '), history.write(subtract), history.write('\n')
            history.close()

            response = input("Would you like to repeat SUBTRACTION?('yes' or 'no'): ")

            while response == 'yes':
                print('\n')
                print(BOLD+DARKCYAN+BORDER+END)

                input1 = input("Enter your first number:   ")
                input2 = input("Enter your second number:   ")
                print('\n')

                subtract = Cal1.subtraction(input1, input2)
                print(input1, " - ", input2, " = ", subtract)
                print('\n')

                history = open('History.txt', 'a')
                history.write(input1),history.write(' - '), history.write(input2), history.write(' = '), history.write(subtract), history.write('\n')
                history.close()

                response = input("Would you like to repeat SUBTRACTION?('yes' or 'no'): ")
    
        elif operation.lower() == 'multiplication':
            input1 = input("Enter your first number:   ")
            input2 = input("Enter your second number:   ")
            product = Cal1.multiplication(input1, input2)
            print(input1, " x ", input2, " = ", product)
            print('\n')

            history = open('History.txt', 'a')
            history.write(input1),history.write(' x '), history.write(input2), history.write(' = '), history.write(product), history.write('\n')
            history.close()

            response = input("Would you like to repeat MULTIPLICATION?('yes' or 'no'): ")

            while response == 'yes':
                print('\n')
                print(BOLD+DARKCYAN+BORDER+END)

                input1 = input("Enter your first number:   ")
                input2 = input("Enter your second number:   ")

                print('\n')

                product = Cal1.multiplication(input1, input2)
                print(input1, " x ", input2, " = ", product)
                print('\n') 

                history = open('History.txt', 'a')
                history.write(input1),history.write(' x '), history.write(input2), history.write(' = '), history.write(product), history.write('\n')
                history.close()

                response = input("Would you like to repeat MULTIPLICATION?('yes' or 'no'): ")
        
        elif operation.lower() == 'division':
            input1 = input("Enter your first number:   ")
            input2 = input("Enter your second number:   ")

            print('\n')

            qoutient = Cal1.division(input1, input2)
            print(input1, " / ", input2, " = ", qoutient)
            print('\n')

            history = open('History.txt', 'a')
            history.write(input1),history.write(' / '), history.write(input2), history.write(' = '), history.write(qoutient), history.write('\n')
            history.close()

            response = input("Would you like to repeat DIVISION?('yes' or 'no'): ")

            while response == 'yes':

                print('\n')
                print(BOLD+DARKCYAN+BORDER+END)

                input1 = input("Enter your first number:   ")
                input2 = input("Enter your second number:   ")

                print('\n')

                qoutient = Cal1.division(input1, input2)
                print(input1, " / ", input2, " = ", qoutient)
                print('\n')

                history = open('History.txt', 'a')
                history.write(input1),history.write(' / '), history.write(input2), history.write(' = '), history.write(qoutient), history.write('\n')
                history.close()

                response = input("Would you like to repeat DIVISION?('yes' or 'no'): ")
    
        else:
            raise Exception(RED+"PLEASE CHOOSE FROM THE FOLLOWING CHOICES ONLY.")
        
        print(BOLD+DARKCYAN+BORDER+END)
        print(BOLD+DARKCYAN+BORDER+END)

        print('\n')
        response = input("Would you like to continue using Calculator 1?('yes' or 'no'): ")

print('\n')
print(BOLD+RED+"Turning off Calculator 1..."+END)
print('\n')
print(BOLD+PURPLE+BORDER+END)
print(BOLD+PURPLE+BORDER+END)



Cal2 = Calculator_Override2() 
response = input("Would you like to open Calculator 2?('yes' or 'no'): ")

if response == 'yes':
    with open('History.txt', 'a') as History2:
        History2.write('\n')
        History2.write("Calculator 2 History"), History2.write('\n')
    
    while response == 'yes':
        operation = input("Which operation ('addition', 'subtraction', 'multiplication' or 'division') would you like to use?: ")
        
        print('\n')
        print(BOLD+DARKCYAN+BORDER+END)
        print('\n')

        if operation.lower() == 'addition':

            input1 = input("Enter your first number:   ")
            input2 = input("Enter your second number:   ")
            print('\n')
            add = Cal2.addition(input1, input2)
            print(input1, " + ", input2, " = ", add)
            print('\n')

            history = open('History.txt', 'a')
            history.write(input1),history.write(' + '), history.write(input2), history.write(' = '), history.write(add), history.write('\n')
            history.close()

            response = input("Would you like to repeat ADDITION?('yes' or 'no'): ")
            

            while response == 'yes':
                print('\n')
                print(BOLD+PURPLE+BORDER+END)

                input1 = input("Enter your first number:   ")
                input2 = input("Enter your second number:   ")

                print('\n')

                add = Cal2.addition(input1, input2)
                print(input1, " + ", input2, " = ", add)
                
                print('\n')

                history = open('History.txt', 'a')
                history.write(input1),history.write(' + '), history.write(input2), history.write(' = '), history.write(add), history.write('\n')
                history.close()

                response = input("Would you like to repeat ADDITION?('yes' or 'no'): ") 

        elif operation.lower() == 'subtraction':
            input1 = input("Enter your first number:   ")
            input2 = input("Enter your second number:   ")

            print('\n')

            subtract = Cal2.subtraction(input1, input2)
            print(input1, " - ", input2, " = ", subtract)
            print('\n')

            history = open('History.txt', 'a')
            history.write(input1),history.write(' - '), history.write(input2), history.write(' = '), history.write(subtract), history.write('\n')
            history.close()
            response = input("Would you like to repeat SUBTRACTION?('yes' or 'no'): ")

            while response == 'yes':
                print('\n')
                print(BOLD+PURPLE+BORDER+END)

                input1 = input("Enter your first number:   ")
                input2 = input("Enter your second number:   ")
                print('\n')

                subtract = Cal2.subtraction(input1, input2)
                print(input1, " - ", input2, " = ", subtract)
                print('\n')

                history = open('History.txt', 'a')
                history.write(input1),history.write(' - '), history.write(input2), history.write(' = '), history.write(subtract), history.write('\n')
                history.close()

                response = input("Would you like to repeat SUBTRACTION?('yes' or 'no'): ")
    
        elif operation.lower() == 'multiplication':
            input1 = input("Enter your first number:   ")
            input2 = input("Enter your second number:   ")
            product = Cal2.multiplication(input1, input2)
            print(input1, " x ", input2, " = ", product)
            print('\n')

            history = open('History.txt', 'a')
            history.write(input1),history.write(' x '), history.write(input2), history.write(' = '), history.write(product), history.write('\n')
            history.close()

            response = input("Would you like to repeat MULTIPLICATION?('yes' or 'no'): ")

            while response == 'yes':
                print('\n')
                print(BOLD+PURPLE+BORDER+END)

                input1 = input("Enter your first number:   ")
                input2 = input("Enter your second number:   ")

                print('\n')

                product = Cal2.multiplication(input1, input2)
                print(input1, " x ", input2, " = ", product)
                print('\n') 

                history = open('History.txt', 'a')
                history.write(input1),history.write(' x '), history.write(input2), history.write(' = '), history.write(product), history.write('\n')
                history.close()

                response = input("Would you like to repeat MULTIPLICATION?('yes' or 'no'): ")
        
        elif operation.lower() == 'division':
            input1 = input("Enter your first number:   ")
            input2 = input("Enter your second number:   ")

            print('\n')

            qoutient = Cal2.division(input1, input2)
            remainder = Cal2.div_rem(input1, input2)
            print(input1, " / ", input2, " = ", qoutient, " Remainder ", remainder)
            print('\n')

            history = open('History.txt', 'a')
            history.write(input1),history.write(' / '), history.write(input2), history.write(' = '), history.write(qoutient), history.write(" Remainder "),history.write(remainder),  history.write('\n')
            history.close()

            response = input("Would you like to repeat DIVISION?('yes' or 'no'): ")

            while response == 'yes':

                print('\n')
                print(BOLD+PURPLE+BORDER+END)

                input1 = input("Enter your first number:   ")
                input2 = input("Enter your second number:   ")

                print('\n')

                qoutient = Cal2.division(input1, input2)
                remainder = Cal2.div_rem(input1, input2)
                print(input1, " / ", input2, " = ", qoutient, " Remainder ", remainder)
                print('\n')

                history = open('History.txt', 'a')
                history.write(input1),history.write(' / '), history.write(input2), history.write(' = '), history.write(qoutient), history.write(" Remainder "),history.write(remainder),  history.write('\n')
                history.close()

                response = input("Would you like to repeat DIVISION?('yes' or 'no'): ")
    
        else:
            raise Exception(RED+"PLEASE CHOOSE FROM THE FOLLOWING CHOICES ONLY.")
        
        print(BOLD+PURPLE+BORDER+END)
        print(BOLD+PURPLE+BORDER+END)

        print('\n')
        response = input("Would you like to continue using Calculator 2?('yes' or 'no'): ")


print('\n')
print(BOLD+RED+"Turning off Calculator 2..."+END)
print('\n')

print(BOLD+BORDER+END)
print(BOLD+BORDER+END)